lib/media.js

```diff
@@ -160,91 +159,86 @@ function injectTags(fileArray, site, tag) {
+  // modern behavior: compile all used variations
+  } else if (state._usedVariations) {
+    cssFilePaths = state._usedVariations
+      .map(variation => {
+        const styleguideVariationPath = path.join(assetDir, 'css', `${variation}.${siteStyleguide}.css`);
+
+        return files.fileExists(styleguideVariationPath) ?
+          styleguideVariationPath :
+          path.join(assetDir, 'css', `${variation}._default.css`);
+      });
+  }
 
-/**
- * @param {Array} list
- * @param {function} fn
- * @returns {Array}
- */
-function flatMap(list, fn) {
-  return _.filter(_.flattenDeep(_.map(list, fn)), _.identity);
+  return cssFilePaths
+    .filter(files.fileExists)
+    .map(pathJoin(assetHost, assetPath, assetDir));
 }
 
 /**
- * Get the proper styles for the component. If a styleguide is configured, use
- * that css. Defaults to the site slug
+ * Get the proper scripts for the component
  *
  * @param {string} name
  * @param {object} site
  * @returns {array}
  */
-function getStyles(name, site) {
-  const siteStyleguide = site && site.styleguide,
-    siteSlug = site && site.slug,
+
+function getScriptFiles(state) {
+  const site = state.locals.site,
+    slug = site.slug,
     assetDir = site && site.assetDir || MEDIA_DIRECTORY,
     assetPath = site && site.assetPath || '',
     assetHost = site && site.assetHost || '';
 
-  let availableVariations, styleguidePath, cssFilePaths;
-
-  if (!siteStyleguide) {
-    // legacy behavior: load <component>.css and <component>.<site>.css
-    cssFilePaths = [
-      path.join(assetDir, 'css', `${name}.css`),
-      path.join(assetDir, 'css', `${name}.${siteSlug}.css`)
-    ];
-  } else {
-    // styleguides: load <component>.<styleguide>.css and <component>_<variation>.<styleguide>.css
-    // note: this also loads default styles if the site styleguide doesn't have styles for a certain component / variation
-    availableVariations = styleguide.getVariations(siteStyleguide);
-    styleguidePath = path.join(assetDir, 'css', `${name}.${siteStyleguide}.css`);
-    cssFilePaths = files.fileExists(styleguidePath) ? [styleguidePath] : [path.join(assetDir, 'css', `${name}._default.css`)]; // fall back to default styleguide
-
-    // if there are variations available for the component, add them to the list
-    // of file paths we want
-    if (availableVariations[name]) {
-      availableVariations[name].forEach((variation) => {
-        let styleguideVariationPath = path.join(assetDir, 'css', `${variation}.${siteStyleguide}.css`);
-
-        cssFilePaths.push(files.fileExists(styleguideVariationPath) ? styleguideVariationPath : path.join(assetDir, 'css', `${variation}._default.css`));
-      });
-    }
-  }
-
-  return _.map(_.filter(cssFilePaths, files.fileExists), pathJoin(assetHost, assetPath, assetDir));
+  return state._components
+    .reduce((prev, name) => {
+      prev.push(
+        path.join(assetDir, 'js', `${name}.js`),
+        path.join(assetDir, 'js', `${name}.${slug}.js`),
```

Comment:
don't think we have site-specific component js?

Reply:
This looks like it was pulled from the previous version that also got three js files for each component https://github.com/clay/amphora-html/blob/v3.3.0/lib/media.js#L192

I can't necessarily speak to why it was setup that way originally, but I suspect it was included here for compatibility reasons